bundles/binding/org.openhab.binding.expire/src/main/java/org/openhab/binding/expire/internal/ExpireGenericBindingProvider.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.expire.internal;
+
+import java.time.Duration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openhab.binding.expire.ExpireBindingProvider;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TypeParser;
+import org.openhab.core.types.UnDefType;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ *
+ * @author Michael Wyraz
+ * @since 1.9.0
+ */
+public class ExpireGenericBindingProvider extends AbstractGenericBindingProvider implements ExpireBindingProvider {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getBindingType() {
+        return "expire";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+        // valid for any item type
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void processBindingConfiguration(String context, Item item, String bindingConfig)
+            throws BindingConfigParseException {
+        super.processBindingConfiguration(context, item, bindingConfig);
+        ExpireBindingConfig config = new ExpireBindingConfig();
+
+        String[] durationAndState = bindingConfig.split(",", 2);
+
+        Duration expiresAfter = parseDuration(durationAndState[0].trim());
+        config.expiresAfterMs = expiresAfter.toMillis();
+        config.expiresAfterAsText = formatDuration(expiresAfter);
+
+        if (durationAndState.length > 1) {
+            // use 2nd parameter as state. defaults to UNDEF
+            config.expiredState = TypeParser.parseState(item.getAcceptedDataTypes(), durationAndState[1].trim());
```

Comment:
The `TypeParser` can return `null`, so I think that warrants throwing a `BindingConfigParseException` that contains an explanation like "The given string '{}' could not be used for any accepted data type for item {}."